prereise/gather/griddata/hifld/data_process/tests/test_transmission.py

```diff
@@ -185,71 +180,49 @@ def test_create_transformers():
         },
         dtype="float",
     )
-    expected_transformers = pd.DataFrame(
-        {"from_bus_id": [0, 2, 4, 5], "to_bus_id": [1, 3, 5, 6]}
-    )
-    transformers = create_transformers(bus)
-    assert_frame_equal(transformers, expected_transformers)
-
-
-def test_estimate_branch_impedance_lines():
-    resistance = 0.01
-    reactance = 0.1
-    fake_lines = [Mock(series_impedance=(resistance + 1j * reactance))] * 3
-    branch = pd.DataFrame(
-        {"VOLTAGE": [69, 70, 345], "type": ["Line"] * 3, "line_object": fake_lines}
+    transformer_designs = pd.DataFrame(
+        {
+            "x": [0.19, 0.07, 0.044, 0.018],
+            "r": [0.01, 0.001, 0.001, 5e-4],
+            "MVA": [60, 250, 280, 580],
+        },
+        index=pd.MultiIndex.from_tuples([(69, 115), (69, 345), (115, 230), (230, 345)]),
     )
-    x = estimate_branch_impedance(branch.iloc[0], pd.Series())
-    assert x == reactance / (69**2 / const.s_base)
-    x = estimate_branch_impedance(branch.iloc[1], pd.Series())
-    assert x == reactance / (70**2 / const.s_base)
-    x = estimate_branch_impedance(branch.iloc[2], pd.Series())
-    assert x == reactance / (345**2 / const.s_base)
-
-
-def test_estimate_branch_impedance_transformers():
-    transformers = pd.DataFrame(
-        {"from_bus_id": [0, 1, 2], "to_bus_id": [1, 2, 3], "type": ["Transformer"] * 3}
+    lines = pd.DataFrame(
+        [
+            # One branch to low-voltage side of buses (0, 1) transformer
+            {"from_bus_id": 100, "to_bus_id": 0, "rateA": 75},
+            # Two branches to high-voltage side of buses (0, 1) transformer
+            {"from_bus_id": 1, "to_bus_id": 101, "rateA": 100},
+            {"from_bus_id": 1, "to_bus_id": 102, "rateA": 300},
+            # Three branches to low-voltage side of buses (2, 3) transformer
+            {"from_bus_id": 103, "to_bus_id": 2, "rateA": 50},
+            {"from_bus_id": 104, "to_bus_id": 2, "rateA": 100},
+            # One branch to high-voltage side of buses (2, 3) transformer
+            {"from_bus_id": 3, "to_bus_id": 105, "rateA": 300},
+            # One branch to each side of buses (5, 6) transformer
+            # The second of these is also on the high-voltage side of buses (4, 5) xfmr
```

Comment:
We don't have any lines on the low-voltage side of buses (4, 5), right?

Reply:
Correct. This is a situation that I don't think we'd ever encounter in our current HIFLD grid-building, since we only add buses where there's a line of that voltage connected to the substation, but the code logic is resilient to this edge case since it uses `int(required_power / single_transformer_power) + 1` so we get `int(0 / x) + 1 = 1` transformer between these two buses.